src/Http/Headers/src/HeaderUtilities.cs

```diff
@@ -696,25 +702,26 @@ public static StringSegment EscapeAsQuotedString(StringSegment input)
             // By calling this, we know that the string requires quotes around it to be a valid token.
             var backSlashCount = CountAndCheckCharactersNeedingBackslashesWhenEncoding(input);
 
-            var stringBuilder = new InplaceStringBuilder(input.Length + backSlashCount + 2); // 2 for quotes
-            stringBuilder.Append('\"');
+            return string.Create(input.Length + backSlashCount + 2, input, (span, segment) => {
```

Comment:
You dropped the comment that the +2 was for the quotes.

Reply:
```suggestion
            // 2 for quotes
            return string.Create(input.Length + backSlashCount + 2, input, (span, segment) => {
```